app/Http/Controllers/Api/Auth/AuthController.php

```diff
@@ -65,9 +68,17 @@ public function register(RegisterHttpRequest $httpRequset, RegisterUserAction $a
         }
     }
 
-    public function me()
+    public function me(GetUserAction $action)
     {
-        return $this->successResponse(AuthPresenter::presentUser(Auth::user()));
+        try {
+            $response = $action->execute(new GetUserRequest(Auth::id()));
+
+            return $this->successResponse(AuthPresenter::presentUser($response));
+        } catch (InvalidUserDataException $exception) {
+            return $this->errorResponse(AuthPresenter::presentError($exception), 400);
+        } catch (\Exception $exception) {
```

Comment:
No need to catch global \Exception. Just your own InvalidUserDataException is ok.